homeassistant/components/tradfri/__init__.py

```diff
@@ -23,28 +22,32 @@
 from homeassistant.helpers.event import async_track_time_interval
 
 from .const import (
-    ATTR_TRADFRI_GATEWAY,
-    ATTR_TRADFRI_GATEWAY_MODEL,
-    ATTR_TRADFRI_MANUFACTURER,
+    _LOGGER,
     CONF_GATEWAY_ID,
     CONF_IDENTITY,
     CONF_KEY,
     COORDINATOR,
     COORDINATOR_LIST,
     DOMAIN,
+    FACTORY,
     KEY_API,
-    PLATFORMS,
-    SIGNAL_GW,
-    TIMEOUT_API,
 )
 from .coordinator import TradfriDeviceDataUpdateCoordinator
 
-_LOGGER = logging.getLogger(__name__)
-
-FACTORY = "tradfri_factory"
-LISTENERS = "tradfri_listeners"
-
+ATTR_TRADFRI_GATEWAY = "Gateway"
+ATTR_TRADFRI_GATEWAY_MODEL = "E1526"
+ATTR_TRADFRI_MANUFACTURER = "IKEA of Sweden"
 CONFIG_SCHEMA = cv.removed(DOMAIN, raise_if_present=False)
+LISTENERS = "tradfri_listeners"
```

Comment:
This constant can be replaced by using `config_entry.async_on_unload` instead of storing unsubscribe callbacks in hass.data.